### ISSUE TITLE
The question about training.

### ISSUE THREAD

**Issue by Casit-ARIS-WQL** [bymaths/probabilistic_triangulation#9]
Hello, your work is great. When I reproduced the experiment, I found that the experimental results were quite different from those in the paper. Could you provide more training details? (I trained the model on 2 RTX 4090)

**Comment by bymaths** [comment]
Please send the loss curve and indicators

**Comment by Casit-ARIS-WQL** [comment]
![image](https://github.com/bymaths/probabilistic_triangulation/assets/141551731/5db690d0-c05c-473d-94be-b75ee87a809e)
![image](https://github.com/bymaths/probabilistic_triangulation/assets/141551731/4174c294-5fff-408a-ba6e-20ac982ba790)

**Comment by bymaths** [comment]
1. What is the horizontal axis? If it is step, the training steps are too few; if it is epoch, the training is too slow, which is abnormal.
2. Try to train 2D first, then lock 2D and then train 3D. Directly training 3D may indeed fall into the local optimum

**Comment by bymaths** [comment]
1. What is the horizontal axis? If it is step, the training steps are too few; if it is epoch, the training is too slow, which is abnormal.
2. Try to train 2D first, then freeze 2D and train 3D. Directly training 3D may indeed fall into the local optimum(because human3.6m is too small)

**Comment by Casit-ARIS-WQL** [comment]
I use the "backbone.pth" you provide for 2D and train 3D, the experimental details are shown as follow:
1. (2 RTX 4090) train_batch_size: 32, test_batch_size: 16, learning_rate: 0.0001 (which is 0.00001 in your code), lr_scheduler: lr*0.9 for 30 epoch, train_logs_file:events.out.tfevents.1717400552.4090-171-train, test_logs_file:events.out.tfevents.1717400552.4090-171-test.
2. (1 RTX 4090) train_batch_size: 16, test_batch_size: 8, learning_rate: 0.0001 (which is 0.00001 in your code), lr_scheduler: lr*0.9 for 30 epoch, train_logs_file: events.out.tfevents.1717400631.4090-171-train, test_logs_file: events.out.tfevents.1717400631.4090-171-test.
3. I try to conduct the experiment with the parameters in your code, but the experimental results are worse.
[logs_file.zip](https://github.com/user-attachments/files/15573525/logs_file.zip), the horizontal axis is epoch.
 The experimental results were still quite different from those in the paper, could you help me?

**Comment by bymaths** [comment]
Very strange result, only by randomly outputting the training estimate can we get the result of "x3d mae 60". I guess there is an error in the calculation of loss between target and prediction.
But I am very busy recently, so I may have to wait until the weekend to check if there is a problem with the code.
You can try to check quickly, what is the result of running metrics directly with the pre-trained 3D pth? If it is not around 27mm, it is probably a problem with the calculation of loss. If it is the same as in the README, it may be a problem with other code in the training part.

**Comment by Casit-ARIS-WQL** [comment]
Your pre-trained 3D pth is 27mm, I will also check the train code. Looking forward to your reply, thanks.
